### PR TITLE
CI: publish to test.pypi.org, normal commit building CI, release CI part 2

### LINKED ISSUE #10: Publish new release and setup CI
changes made so need to publish new release and setup CI

### PR DESCRIPTION
**Pull request by Zeroto521**
closes #10

### CODE REVIEW

**Comment by Zeroto521** [comment]
Ready to go.
But I'm not sure why this new ci (`.github/workflows/sdist.yaml`) doesn't run.
Is it a setting problem?

**Comment by taooceros** [comment]
Maybe we need to merge first so the ci will run.

File: .github/workflows/sdist.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.6"
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
+    steps:
+      - uses: actions/checkout@v2
```
Comment (taooceros) [inline .github/workflows/sdist.yaml:30]: Let's use the latest v3

Comment (Zeroto521) [reply .github/workflows/sdist.yaml:30]: I will add dependably in another pr. Let magic fight with magic. Not only does that one need to update.

File: .github/workflows/sdist.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      - name: Install from sdist
+        run: |
+          pip list
+          python -m pip install dist/*.gz
+
+      - name: Import flowlauncher
+        run: |
+          python -V
+          python -c "from flowlauncher import __version__; print(__version__);"
+          python -c "import pprint; from flowlauncher._version import get_versions; pprint.pprint(get_versions());"
```
Comment (taooceros) [inline .github/workflows/sdist.yaml:56]: is these few lines only printed to console as information?

File: .github/workflows/sdist.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          conda info
+          conda list
+
+      - name: Publish to TestPyPI
+        if: ${{ github.event_name == 'push' }}
+        uses: pypa/gh-action-pypi-publish@v1.4.2
+        with:
+          user: __token__
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
```
Comment (taooceros) [inline .github/workflows/sdist.yaml:66]: why there's a legacy here?

Comment (Zeroto521) [reply .github/workflows/sdist.yaml:66]: This is the route where we upload package to testpypi.

Comment (taooceros) [reply .github/workflows/sdist.yaml:66]: Yeah i see. I just wondering whether the legacy means it is depreciated?

Comment (Zeroto521) [reply .github/workflows/sdist.yaml:66]: official recommend this.

Comment (taooceros) [reply .github/workflows/sdist.yaml:66]: Okok thanks

File: .github/workflows/sdist.yaml (hunk 2 above)
Comment (Zeroto521) [reply .github/workflows/sdist.yaml:56]: Github Action outputs. For us to debug version and trace commit.
`get_versions` will output like the following. And `__version__` is the `get_versions()['version]` acutally.
```json
{'date': '2022-08-16T17:03:51+0800',
 'dirty': False,
 'error': None,
 'full-revisionid': '0b8071bd1cb435eb2f3e995f31cdefd7a958e5ed',
 'version': '0.post57'}
```

**Review by taooceros — APPROVED** [review]

**Pull request merged**